gradle-runner-agent/src/main/scripts/init.gradle

```diff
@@ -811,13 +811,34 @@ class TestTaskWrapper {
     }
     return this
   }
+
   TestTaskWrapper setTmpDirectory(String path) {
     if (testTask instanceof Test &&
         path != null && path != "") {
       testTask.systemProperty("java.io.tmpdir", path)
     }
     return this
   }
+
+  /**
+   * Creates an instance of the {@link TestTaskWrapper} for every instance of the Test in the TaskCollection.
+   * Attempts to lazily evaluate the Closure action against all instances lazily if the version of Gradle supports it.
+   */
+  static void forAllTestTasks(TaskCollection taskCollection, Closure action) {
+    if (taskCollection.metaClass.respondsTo(taskCollection, 'configureEach')) {
+      // Use lazy task configuration:
+      // https://docs.gradle.org/current/userguide/task_configuration_avoidance.html
+      taskCollection.configureEach { task ->
+        TestTaskWrapper wrapper = new TestTaskWrapper(task)
+        action(wrapper)
+      }
+    } else {
+      // Fallback
+      taskCollection
+          .collect { new TestTaskWrapper(it) }
+          .each(action)
```

Comment:
@madlexa This will only configure tasks that exist when the `forAllTestTasks` method is called.
Moving this to use `TaskCollection::all` would mean that all `Test` tasks get this `TestTaskWrapper` created for them.

Do you want me to make a change to using `all` or is there a specific reason why `collect` was used previously?

Reply:
What do you mean? Extension method for TaskCollection? This is possible, but you only need to come up with a new method name e.g. tcLazyEach or tcConfigureEach.

Reply:
`all` is a method on `TaskCollection` that has existed for quite a while.

Reply:
https://docs.gradle.org/current/javadoc/org/gradle/api/DomainObjectCollection.html#all-groovy.lang.Closure-

Reply:
Yes! Cool! Good idea!

Reply:
According to the JetBrains security policy, we only accept PRs from users who have signed the [JetBrains CLA](https://www.jetbrains.com/agreements/cla/). If you consider contributing to this project or other JB projects in the future, could you please sign and send it to us, so I can accept your PR?